server/src/main/java/org/elasticsearch/cluster/metadata/DataStream.java

```diff
@@ -1396,4 +1309,139 @@ private static Instant getTimestampFromParser(BytesReference source, XContentTyp
     public static Instant getCanonicalTimestampBound(Instant time) {
         return time.truncatedTo(ChronoUnit.SECONDS);
     }
+
+    public static class Builder {
+        private LongSupplier timeProvider;
+        private String name;
+        private List<Index> indices;
+        private long generation;
```

Comment:
If I make a new instance builder and set no values on anything and try to call build(), would we get a valid data stream object out of it? What would the generation be in that case?

Reply:
Discussed and addressed in the comment below -- resolving this one.